components/script/dom/window.rs

```diff
@@ -1505,34 +1638,48 @@ impl Window {
 
         for image in complete.pending_images {
             let id = image.id;
-            let js_runtime = self.js_runtime.borrow();
-            let js_runtime = js_runtime.as_ref().unwrap();
-            let node = unsafe { from_untrusted_node_address(js_runtime.rt(), image.node) };
+            let node = match &*self.current_state.borrow_mut() {
+                WindowState::Zombie(_) => {
+                    unreachable!("Window should be alive when force_reflow is called")
```

Comment:
Would you think it's good to have a `is_zombie` function and assert with `is not zombie` before running this for loop 👀?

```rust
let current_state = self.current_state.borrow_mut();

assert!(!current_state.is_zombie());

for image in complete.pending_images {
  let id = image.id;
  let node = current_state.data().js_runtime.as_ref().unwrap(); // this is just a pseudo code 👀
  ...
  let mut images = current_state.data().pending_layout_images.borrow_mut();
  ...
}
```

Reply:
Thanks for the suggestion, I think I prefer to match on it and do the data handling/assertion in the match, versus separating it in a `is_zombie` ad `data` method, since it could be quite easy to forget the initial `assert!(!current_state.is_zombie());`, and probably the implementation of `data` would need to require another `assert!(!current_state.is_zombie());`, or otherwise handle the different state. Also, the zombie state also has some data, but it's not the same data, so I think the user of `data` would have to match on the return value anyhow.

Reply:
> Also, the zombie state also has some data, but it's not the same data, so I think the user of data would have to match on the return value anyhow.

I see! Sounds like matching on the state is the wise way and won't confuse other developers 😇